azurerm/resource_arm_virtual_machine.go

```diff
@@ -630,6 +644,12 @@ func resourceArmVirtualMachineCreateUpdate(d *schema.ResourceData, meta interfac
 			properties.DiagnosticsProfile = diagnosticsProfile
 		}
 	}
+	if _, ok := d.GetOk("additional_capabilities"); ok {
+		additionalCapabilities := expandAzureRmVirtualMachineAdditionalCapabilities(d)
+		if additionalCapabilities != nil {
+			properties.AdditionalCapabilities = additionalCapabilities
```

Comment:
since expand returns nil (and the SDK has `omitempty` we can just directly assign this here without the nil-check:

```suggestion
		properties.AdditionalCapabilities = expandAzureRmVirtualMachineAdditionalCapabilities(d)
```

Reply:
optimised code according to your suggestion.
Thanks